pkg/rke2/rke2.go

```diff
@@ -166,5 +175,150 @@ func setup(clx *cli.Context, cfg Config) error {
 	}
 	executor.Set(&sp)
 
+	disabledItems := map[string]bool{
+		"kube-apiserver":          disableAPIServer,
+		"kube-scheduler":          disableScheduler,
+		"kube-controller-manager": disableControllerManager,
+		"etcd":                    disableETCD,
+	}
+	return removeOldPodManifests(dataDir, disabledItems)
+}
+
+func podManifestsDir(dataDir string) string {
+	return filepath.Join(dataDir, "agent", config.DefaultPodManifestPath)
+}
+
+func removeOldPodManifests(dataDir string, disabledItems map[string]bool) error {
+	var kubeletStandAlone bool
+
+	kubeletErr := make(chan error)
+	containerdErr := make(chan error)
+
+	ctx, cancel := context.WithTimeout(context.Background(), (5 * time.Minute))
+	defer cancel()
+
+	manifestDir := podManifestsDir(dataDir)
+	for component, disabled := range disabledItems {
+		if disabled {
+			manifestName := filepath.Join(manifestDir, component+".yaml")
+			if _, err := os.Stat(manifestName); err == nil {
+				kubeletStandAlone = true
+				if err := os.Remove(manifestName); err != nil {
+					return err
+				}
+			}
+		}
+	}
+	if kubeletStandAlone {
+		kubeletCmd := exec.Command("kubelet")
+		containerdCmd := exec.Command("containerd")
+
+		tmpAddress := filepath.Join(os.TempDir(), "containerd.sock")
+
+		// start containerd
+		go startContainerd(ctx, dataDir, containerdErr, tmpAddress, containerdCmd)
+		// start kubelet
+		go startKubelet(ctx, dataDir, kubeletErr, tmpAddress, kubeletCmd)
+
+		// check for any running containers from the disabled items list
+		go checkForRunningContainers(ctx, tmpAddress, disabledItems, kubeletErr, containerdErr)
+
+		for {
+			time.Sleep(5 * time.Second)
+			select {
+			case err := <-kubeletErr:
+				logrus.Infof("kubelet Exited: %v, exiting Containerd", err)
+				// exits containerd if kubelet exits
+				containerdCmd.Process.Kill()
+				kubeletCmd.Process.Kill()
+
+			case err := <-containerdErr:
+				logrus.Infof("Containerd Exited: %v, exiting kubelet", err)
+				kubeletCmd.Process.Kill()
+				containerdCmd.Process.Kill()
+
+			case <-ctx.Done():
+				logrus.Info("Timeout reached, exiting kubelet and containerd")
+				kubeletCmd.Process.Kill()
+				containerdCmd.Process.Kill()
+			}
+			break
+		}
+	}
+
 	return nil
 }
+
+func startKubelet(ctx context.Context, dataDir string, errChan chan error, tmpAddress string, cmd *exec.Cmd) {
```

Comment:
You're not actually using the Context that you pass in to this or startContainerd - if they don't use the context don't bother passing it in.

Reply:
This needs addressed, other than that LGTM